x/ibc-transfer/handler_test.go

```diff
@@ -3,309 +3,64 @@ package transfer_test
-	err = suite.chainA.App.ScopedTransferKeeper.ClaimCapability(suite.chainA.GetContext(), cap, capName)
-	suite.Require().Nil(err, "transfer module could not claim capability")
-
-	ctx := suite.chainA.GetContext()
-	msg := types.NewMsgTransfer(testPort1, testChannel1, testPrefixedCoins2, testAddr1, testAddr2.String(), 110, 0)
-	res, err := handler(ctx, msg)
-	suite.Require().Error(err)
-	suite.Require().Nil(res, "%+v", res) // channel does not exist
+	clientA, clientB, _, _, channelA, channelB := suite.coordinator.Setup(suite.chainA, suite.chainB)
+	handlerA := ibctransfer.NewHandler(suite.chainA.App.TransferKeeper)
 
-	// Setup channel from A to B
-	suite.chainA.CreateClient(suite.chainB)
-	suite.chainA.createConnection(testConnection, testConnection, testClientIDB, testClientIDA, connectiontypes.OPEN)
-	suite.chainA.createChannel(testPort1, testChannel1, testPort2, testChannel2, channeltypes.OPEN, channeltypes.ORDERED, testConnection)
+	coinToSendToB := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelB.PortID, channelB.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
 
-	res, err = handler(ctx, msg)
-	suite.Require().Error(err)
-	suite.Require().Nil(res, "%+v", res) // next send sequence not found
+	// send from chainA to chainB
+	msg := types.NewMsgTransfer(channelA.PortID, channelA.ID, coinToSendToB, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), 110, 0)
 
-	nextSeqSend := uint64(1)
-	suite.chainA.App.IBCKeeper.ChannelKeeper.SetNextSequenceSend(ctx, testPort1, testChannel1, nextSeqSend)
-	res, err = handler(ctx, msg)
-	suite.Require().Error(err)
-	suite.Require().Nil(res, "%+v", res) // sender has insufficient coins
-
-	_ = suite.chainA.App.BankKeeper.SetBalances(ctx, testAddr1, testCoins)
-	res, err = handler(ctx, msg)
+	res, err := handlerA(suite.chainA.GetContext(), msg)
 	suite.Require().NoError(err)
 	suite.Require().NotNil(res, "%+v", res) // successfully executed
 
-	// test when the source is false
-	msg = types.NewMsgTransfer(testPort1, testChannel1, testPrefixedCoins2, testAddr1, testAddr2.String(), 110, 0)
-	_ = suite.chainA.App.BankKeeper.SetBalances(ctx, testAddr1, testPrefixedCoins2)
+	err = suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, []sdk.Msg{msg})
+	suite.Require().NoError(err) // message committed
+
+	handlerB := ibctransfer.NewHandler(suite.chainB.App.TransferKeeper)
 
-	res, err = handler(ctx, msg)
-	suite.Require().Error(err)
-	suite.Require().Nil(res, "%+v", res) // incorrect denom prefix
+	coinToSendBackToA := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelA.PortID, channelA.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
 
-	msg = types.NewMsgTransfer(testPort1, testChannel1, testPrefixedCoins1, testAddr1, testAddr2.String(), 110, 0)
-	suite.chainA.App.BankKeeper.SetSupply(ctx, banktypes.NewSupply(testPrefixedCoins1))
-	_ = suite.chainA.App.BankKeeper.SetBalances(ctx, testAddr1, testPrefixedCoins1)
+	// send from chainB back to chainA
```

Comment:
ah this test should actually fail because the coin is never recv'd and ack'd. I think this can be updated when the fix is posted for handling the source correctly. Probably should create a helper func in ibctesting to handle this

Reply:
can you add this comment to the fix issue?